### PR TITLE
🌊 Only_onnxruntime to false when opt_level > 1

### PR DESCRIPTION
**Pull request by trajepl**
## Describe your changes
1. Only_onnxruntime to false when opt_level > 1
2. set quantization calibration_method to minmax as the bug from ort 1.16.0 https://github.com/microsoft/onnxruntime/issues/17619

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/passes/onnx/transformer_optimization.py [hunk 1]
```diff
@@ -59,8 +59,19 @@ def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigPa
             "only_onnxruntime": PassConfigParam(
                 type_=bool,
                 default_value=False,
-                searchable_values=Boolean(),
-                description="Whether only use onnxruntime to optimize model, and no python fusion.",
+                searchable_values=Conditional(
+                    parents=("opt_level",),
+                    support={
+                        (2,): Boolean([False]),
+                        (0,): Boolean(),
```
Comment (jambayk) [inline olive/passes/onnx/transformer_optimization.py:66]: not an issue but the default value for other params can be set by using https://github.com/microsoft/Olive/blob/main/olive/strategy/search_parameter.py#L127.

Comment (trajepl) [reply olive/passes/onnx/transformer_optimization.py:66]: ahah, thanks! I even think it is a bug where we do not have the default searchable options.

File: olive/strategy/search_parameter.py [hunk 2]
```diff
@@ -80,8 +80,8 @@ class Boolean(Categorical):
     >>> Boolean()
     """
 
-    def __init__(self):
-        super().__init__([True, False])
+    def __init__(self, support: List[bool] = [True, False]):
```
Comment (jambayk) [inline olive/strategy/search_parameter.py:83]: this has the same issue as using mutable objects as default values. 

if we only want one option, we can just use `Categorical([False])` instead. `Boolean` was created as a shorthand for `Categorical([True, False])`

File: examples/bert/bert_ptq_cpu.json [hunk 3]
```diff
@@ -59,6 +60,7 @@
                 "quant_preprocess": true,
                 "per_channel": false,
                 "reduce_range": false,
+                "calibrate_method": "MinMax",
```
Comment (guotuofeng) [inline examples/bert/bert_ptq_cpu.json:63]: add TODO here

Comment (trajepl) [reply examples/bert/bert_ptq_cpu.json:63]: hard to comment for Json, I add some comment here.
image

File: olive/strategy/search_parameter.py (hunk 2 above)
Comment (trajepl) [reply olive/strategy/search_parameter.py]: Fixed.

File: olive/strategy/search_parameter.py [hunk 4]
```diff
@@ -80,7 +80,7 @@ class Boolean(Categorical):
     >>> Boolean()
     """
 
-    def __init__(self):
+    def __init__(self, support: List[bool] = None):
         super().__init__([True, False])
```
Comment (guotuofeng) [inline olive/strategy/search_parameter.py:84]: not pass default arrays, Instead, better pass None and let the Categorical handle Non

Comment (trajepl) [reply olive/strategy/search_parameter.py]: Emm, Boolean() is used to directly create Categorical([True, False]). If we pass None, we can directly use Categorical.

(If the arguments `support` confuses you, I removed in latest commit as it is the zombie code.

Comment (guotuofeng) [reply olive/strategy/search_parameter.py]: I mean 
image

File: olive/strategy/search_parameter.py [hunk 5]
```diff
@@ -313,9 +313,10 @@ def json_to_search_parameter(json: Dict[str, Any]) -> SearchParameter:
     if search_parameter_type == "Categorical":
         return Categorical(json["support"])
     if search_parameter_type == "Conditional" or search_parameter_type == "ConditionalDefault":
-        stop_condition = lambda x: (  # noqa: E731
-            isinstance(x, dict) and x.get("olive_parameter_type") == "SearchParameter"
-        )
+
+        def stop_condition(x):
+            return isinstance(x, dict) and x.get("olive_parameter_type") == "SearchParameter"  # noqa: E731
```
Comment (guotuofeng) [inline olive/strategy/search_parameter.py:318]: > # noqa: E731 [](http://example.com/codeflow?start=94&length=13)

no need this one?

Comment (trajepl) [reply olive/strategy/search_parameter.py:318]: Good catch. This line is auto-formatted from my local linter plugin. But somehow the `noqa` was not removed.

**Review by guotuofeng — APPROVED** [review]

**Pull request merged**